Adds guide about how to revert to a model snapshot (#2156)
elastic/stack-docs: docs/en/stack/ml/anomaly-detection/ml-revert-model-snapshot.asciidoc

[role="xpack"]
[[ml-reverting-model-snapshot]]
= Reverting to a model snapshot

{ml-docs}/ml-ad-run-jobs.html#ml-ad-model-snapshots[Snapshots of the {ml} model] 
for each {anomaly-job} are saved frequently to an internal {es} index to ensure 
resilience. It makes it possible to reset the model to a previous state in case 
of a system failure or if the model changed significantly due to a one-off 
event.

. In {kib}, navigate to **{ml-app} > Anomaly Detection > Jobs**.
. Locate the {anomaly-job} whose model you want to revert in the job table.
. Open the job details and navigate to the **Model Snapshots** tab.
+
--
[role="screenshot"]
image::images/anomaly-job-model-snapshots.jpg[A screenshot of a job with the Model Snapshots tab opened]
--

> . Select the snapshot you want to revert from the list and click the **Revert** 

Reviewer: Not mandatory, but I think this is clearer since "reverting a snapshot" implies you're doing something to the snapshot itself:
```suggestion
. Select a snapshot from the list and click the **Revert** 
```

  icon under **Actions**.

> . Optional: Select if you want to replay the analysis after the revert has been 

Reviewer: > . Optional: Select if you want to replay the analysis after the revert has been 

It might be nice to clarify what is meant by "replay the analysis" since I can't tell from this quick read and scanning the screenshot what that means. Does it mean that it's re-applying analysis that it already did in the past on top of the reverted job?  That seems to contradict the statement at the bottom of the page ("In case of reverting to a snapshot, all anomaly detection results are deleted for the corresponding job after the point when the snapshot was saved.")

  applied. You can decide whether you want the job running in real time. You can 
  also define the time range you don't want the job to analyze, this way you can 
  skip the time of the event you don't want to impact the model behavior. 
+
--
[role="screenshot"]
image::images/revert-model-snapshot.jpg[A screenshot of a revert model snapshot flyout]
--
. Click **Apply**.

> 

Reviewer: I think it might also be a good idea to mention/link to custom rules in this page too. IIRC they are a way to tell the model to avoid updating the model in certain circumstances (and therefore potentially avoiding future situations where you'd need to revert).  There's a blurb about them in https://stack-docs_2156.docs-preview.app.elstc.co/guide/en/machine-learning/master/ml-ad-run-jobs.html#ml-ad-rules

Author: Added in [2381b34](https://github.com/elastic/stack-docs/pull/2156/commits/2381b347b6dbb6617ed11826718ad1e2296b74b9).

  **Actions**.
. Optional: Select if you want to replay the analysis based on the data in your 
  index after the revert has been applied. If you don’t select this option, 
  there will be no {anomaly-detect} results after the snapshot was taken and the 
  job results end at the point in time of the snapshot.If you select this 
  option, you can then optionally select one or both of the next two options:

  * You can select whether you want the job to continue running in real time 
  after the replay, or to just replay existing data after the snapshot.
  * You can select a time range you want to avoid during the replay by declaring 
  a Calendar event. This way, you can skip any problematic time frame that you 
  want the {anomaly-job} to avoid.

+
--
[role="screenshot"]
image::images/revert-model-snapshot.jpg[A screenshot of a revert model snapshot flyout]
--
. Click **Apply**.

TIP: You can use <<ml-ad-rules,custom rules>> to avoid a model being updated in 
  case of a known event you want to exclude from the analysis. Using custom 
  rules might help you to avoid situations where you need to revert to a 
  snapshot.

Alternatively, you can use the 

> {ref}/ml-revert-snapshot.html[Revert model snapshots] API. In this case, you 

Reviewer: This doesn't require capitalization either:
```suggestion
{ref}/ml-revert-snapshot.html[revert model snapshots] API. In this case, you 
```

need to manually close the corresponding job before reverting to the saved 
snapshot.

[NOTE]
===============================

> * In case of reverting to a snapshot, all {anomaly-detect} results are deleted 

Reviewer: I found "in case of reverting to a snapshot" unclear.  Which particular case is it referring to? Only when you do it via API?  Or some other circumstance that I'm missing? If it's applicable to all methods, something like this might be clearer:
```suggestion
* By default, when you revert to a snapshot, all {anomaly-detect} results are deleted 
```
